libfwupdplugin/fu-string.h

```diff
@@ -15,6 +15,13 @@ fu_strpassmask(const gchar *str) G_GNUC_NON_NULL(1);
 gboolean
 fu_strtoull(const gchar *str, guint64 *value, guint64 min, guint64 max, GError **error);
 gboolean
+fu_strtoull_full(const gchar *str,
+		 guint64 *value,
+		 guint64 min,
+		 guint64 max,
+		 guint base,
+		 GError **error);
+gboolean
```

Comment:
with the ABI bump, do we need to keep the two functions?  I know it's more changes to plugins, but it would be cleaner to just keep one I feel like.

Or have helper marcros that specify the common bases like:

`fu_stroull_base_10` and `fu_stoull_base_16`

Reply:
Counterproposal -- what about always specifying the base, i.e. we change all the plugins, but we have a magic `#define FU_STRING_BASE_AUTO 0` (other names welcome) which uses the `0x` prefix for base 16, and base 10 otherwise. Then we don't get the additional symbol and it's all a bit less magic.

Reply:
Sounds good

Reply:
Done.